packages/protocol/contracts/common/GoldToken.sol

```diff
@@ -174,8 +153,8 @@ contract GoldToken is
   function transferFrom(address from, address to, uint256 value) external returns (bool) {
     require(to != address(0), "transfer attempted to reserved address 0x0");
     require(
-      to != address(celoTokenDistributionSchedule),
-      "transfer attempted to reserved celoTokenDistributionSchedule address"
+      to != registry.getAddressForOrDie(CELO_DISTRIBUTION_SCHEDULE_ID),
```

Comment:
Since we are using CELO_DISTRIBUTION_SCHEDULE_ID multiple times should we add it to UsingRegistry ?

Reply:
I ended up using `registry.getAddressForOrDie(CELO_DISTRIBUTION_SCHEDULE_ID)` to optimize for gas cost.

Otherwise, we convert an address to an interface and back to an address.

Reply:
I can change it if you think its necessary.

Reply:
For context, what's the difference in gas cost (if that's easy to calculate). I have no intuition for that. 

Personally, I'd prioritise contract readability over gas efficiency, but that's coming from a place of "maintaining" the contracts, and less of "using" the contracts. So I'll defer judgement to @pahor167 and you on this @soloseng.

Reply:
I'm not sure if it saves gas but I like it because it helps us move in this direction https://github.com/celo-org/celo-monorepo/issues/11151

Reply:
> For context, what's the difference in gas cost (if that's easy to calculate)? I have no intuition for that.

I usually check by running the test in foundry using both methods when I want to have an idea of which would be higher.